frame/balances/src/lib.rs

```diff
@@ -730,6 +730,12 @@ pub mod pallet {
 			let existential_deposit = Self::ed();
 
 			let wipeout = new_free < existential_deposit;
+			debug_assert!(
+				!wipeout || <T::Balance as Zero>::is_zero(&new_free),
```

Comment:
```suggestion
				!wipeout || new_free.is_zero(),
```

Reply:
I agree with Liam here -- this seems like we should use `log::warn` instead of debug.